### ISSUE TITLE
Panic during (*logWriter).WriteFinished

### ISSUE THREAD

**Issue by lwc** [buildkite/ecs-run-task#2]
Just saw this, dumping it here so I don't forget

```
panic: runtime error: invalid memory address or nil pointer dereference
--
  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7e7792]
  |  
  | goroutine 39 [running]:
  | github.com/buildkite/ecs-run-task/runner.(*logWriter).WriteFinished(0xc4201af530, 0xc420498a20, 0xc420270870, 0x0, 0x0)
  | /go/src/github.com/buildkite/ecs-run-task/runner/runner.go:286 +0x162
  | github.com/buildkite/ecs-run-task/runner.(*Runner).Run.func1(0xc4202225a0, 0xc4200fc180, 0xc420017e20, 0x12, 0xc420242028, 0xc420047000, 0xc420242018, 0xc4202420e0, 0x1, 0x1)
  | /go/src/github.com/buildkite/ecs-run-task/runner/runner.go:163 +0x31f
  | created by github.com/buildkite/ecs-run-task/runner.(*Runner).Run
  | /go/src/github.com/buildkite/ecs-run-task/runner/runner.go:134 +0x13d8

```

**Comment by Vektah** [comment]
I wonder if write finished should exist at all?

It requires giving our CI instances permission to write prod logs and doesn't really add much.

**Comment by lox** [comment]
How else do you figure out when the logs are finished?

**Comment by Vektah** [comment]
Ah, so this is a way to work out when you have read enough?

Does this really work? To push to an existing stream you need a sequenceToken, which you get by calling `DescribeLogStreams`. If the call to `DescribeLogStreams` is as eventually consistent as `GetLogEvents` then the order is probably undefined anyway?

**Comment by lox** [comment]
Yeah, there is some blurryness around the edges, but it beats the hell out of "Wait N seconds for output to complete"

**Comment by lox** [comment]
I think this error was caused by `ExitCode` being nil when the task fails before it starts, for instance when you try and execute a command that doesn't exist.

**Comment by lox** [comment]
Closed in #18.
